packages/core/integration-tests/test/integration/resolver-dependency-meta/node_modules/parcel-resolver-meta/index.js

```diff
@@ -0,0 +1,11 @@
+const {Resolver} = require('@parcel/plugin');
+
+module.exports = new Resolver({
+  resolve({dependency, options, filePath}) {
+    if(dependency.moduleSpecifier === "foo"){
+      dependency.meta.customValue = 1234;
```

Comment:
😬 should this be allowed?? I thought meta was immutable... What's the use case for doing this?

Reply:
Marking dependencies which should later be overwritten in a runtime

<img width="610" alt="Bildschirmfoto 2020-08-15 um 23 11 15" src="https://user-images.githubusercontent.com/4586894/90321723-9e9fc200-df4c-11ea-88c9-c8d4b1f0f593.png">

Reply:
Is that to avoid loading the manifest in the runtime as well?

Reply:
That is only a side-effect. 
I did this to differentiate "isExcluded" and "my special import". But in a browser build, nothing can be excluded anyway....

Reply:
I mean assigning to the `manifestRoot` is so that a runtime doesn't need to load the manifest to check if the module specifier is there? Is there a downside to doing that? Seems like we should generally try to avoid mutating things for cacheability reasons... I bet there are lots of other assumptions in the code about that.

Reply:
> Is there a downside to doing that

I wanted to mark some specific excluded dependencies as "special", so I could have done `dependency.meta.somthing = true;`. 

Though again:

> But in a browser build, nothing can be excluded anyway....

<br>

> I mean assigning to the manifestRoot is so that a runtime doesn't need to load the manifest to check if the module specifier is there

Loading a config in the resolver to get the root would work as well, I just thought "instead of `true`, I could pass some information I need anyway".

<br>

> Seems like we should generally try to avoid mutating things for cacheability reasons

The question is: where is it explicitly allowed? Only in the transformer?